### PR TITLE
Removing lenses from integration tests (state, addressPoolGap, balance*)

### PR DESCRIPTION
**Pull request by piotr-iohk**
# Issue Number

#1237

# Overview

- b43b949ffa63acf4e7f8a7d88da0d7df92dc1844
  Remove addressPoolGap lens
  
- 24d318ca75b8eccf732f1b0bfbbf01fde4944e92
  Remove state lens
  
- 446f7812916ca96380fb7f4946a4a057e6f7b634
  Remove balance lenses:  - byronBalanceAvailable  - balanceAvailable  - byronBalanceTotal  - balanceTotal  - balanceReward
  
- 9d2e67981eeb63698dc69994c758013601e8802f
  get rid of duplicate magic constants
  - addressPoolGapMin
  - addressPoolGapMax
  - passphraseMinLength
  - passphraseMaxLength

Also got rid of an extra print statement that slipped through

# Comments

### CODE REVIEW

**Comment by piotr-iohk** [comment]
bors try

**Comment by piotr-iohk** [comment]
@KtorZ, perhaps it could be reviewed at this stage. If OK removing of other lenses can go in next prs.

File: lib/core-integration/src/Test/Integration/Framework/TestData.hs [hunk 1]
```diff
@@ -288,10 +289,10 @@ passphraseMinLength = 10
 passphraseMaxLength :: Int
 passphraseMaxLength = 255
 
-addressPoolGapMin :: Int
+addressPoolGapMin :: Word32
 addressPoolGapMin = 10
```
Comment (KtorZ) [inline lib/core-integration/src/Test/Integration/Framework/TestData.hs]: What about using an `AddressPoolGap` type and relying on `minBound` ?

File: lib/core-integration/src/Test/Integration/Framework/TestData.hs [hunk 2]
```diff
@@ -288,10 +289,10 @@ passphraseMinLength = 10
...
 
-addressPoolGapMax :: Int
+addressPoolGapMax :: Word32
```
Comment (KtorZ) [inline lib/core-integration/src/Test/Integration/Framework/TestData.hs]: What about using an `AddressPoolGap` type and relying on `maxBound` ?

File: lib/core-integration/src/Test/Integration/Scenario/API/Addresses.hs [hunk 3]
```diff
@@ -182,7 +182,9 @@ spec = do
         -- make sure all transactions are in ledger
         rb <- request @ApiWallet ctx
             (Link.getWallet @'Shelley wDest) Default Empty
-        expectEventually ctx (Link.getWallet @'Shelley) balanceAvailable 10 rb
+        expectEventually ctx (Link.getWallet @'Shelley)
+                (#balance . #getApiT . #available) (Quantity 10) rb
+        print rb
```
Comment (KtorZ) [inline lib/core-integration/src/Test/Integration/Scenario/API/Addresses.hs]: An extra `print rb` has sneaked into this code :)

**Review by KtorZ — APPROVED** [review]

File: lib/core-integration/src/Test/Integration/Framework/TestData.hs (hunk 1 above)
Comment (KtorZ) [reply lib/core-integration/src/Test/Integration/Framework/TestData.hs]: done in 9d2e679

File: lib/core-integration/src/Test/Integration/Framework/TestData.hs (hunk 2 above)
Comment (KtorZ) [reply lib/core-integration/src/Test/Integration/Framework/TestData.hs]: done in 9d2e679

**Pull request merged**